usaspending_api/common/helpers/decorators.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from django.conf import settings
+from django.db import connection
+from django.db.utils import OperationalError
+
+from usaspending_api.common.exceptions import EndpointTimeoutException
+
+logger = logging.getLogger(__name__)
+
+
+def set_db_timeout(timeout_in_seconds=settings.DEFAULT_DB_TIMEOUT_IN_SECONDS):
+    """ Decorator used to set the database statement timeout within the Django app scope
+
+        Args:
+            timeout_in_seconds: timeout value to set, in milliseconds
+
+        NOTE:
+            The statement_timeout is only set for this specific connection. Any subsequent request will have a default
+            value of 0. The alteration DOES NOT persist across connections and since each request is a new connection to
+            the database, there is no need to reset the statement_timeout value.
+
+        Examples:
+            @set_db_timeout(test_timeout_in_ms)
+            def func_running_db_call(...):
+                ...
+
+            OR
+
+            @set_db_timeout()  # This will use the default value in settings
```

Comment:
Can that default timeout not be applied globally, _as the default_? Or must every call that needs to timeout have this decorator (with or without args)?

Maybe there's a global patch that can be applied to set a default `statement_timeout` on anything within the app that calls the db?

Reply:
The timeout won't be set unless the decorator is used. So it won't affect anything globally.